VRDR/DeathRecord.cs

```diff
@@ -1466,1339 +1448,663 @@ public string ContributingConditions
-                 {
-                    CauseOfDeathConditionG = CauseOfDeathCondition(6);
-                 }
-                CauseOfDeathConditionG.Onset = new FhirString(value);
-             }
-        }
-
-        /// <summary>Cause of Death Part I Code, Line g.</summary>
-        /// <value>the sixth underlying cause of death coding. A Dictionary representing a code, containing the following key/value pairs:
-        /// <para>"code" - the code</para>
-        /// <para>"system" - the code system this code belongs to</para>
-        /// <para>"display" - a human readable meaning of the code</para>
-        /// </value>
-        /// <example>
-        /// <para>// Setter:</para>
-        /// <para>Dictionary&lt;string, string&gt; code = new Dictionary&lt;string, string&gt;();</para>
-        /// <para>code.Add("code", "example");</para>
-        /// <para>code.Add("system", "example");</para>
-        /// <para>code.Add("display", "example");</para>
-        /// <para>ExampleDeathRecord.CODE1G = code;</para>
-        /// <para>// Getter:</para>
-        /// <para>Console.WriteLine($"\tCause of Death: {ExampleDeathRecord.CODE1G['display']}");</para>
-        /// </example>
-        [Property("CODE1G", Property.Types.Dictionary, "Death Certification", "Cause of Death Part I Code, Line g.", false, "http://build.fhir.org/ig/HL7/vrdr/StructureDefinition-VRDR-Cause-Of-Death-Condition.html", false, 100)]
-        [PropertyParam("code", "The code used to describe this concept.")]
-        [PropertyParam("system", "The relevant code system.")]
-        [PropertyParam("display", "The human readable version of this code.")]
-        [PropertyParam("text", "Additional descriptive text.")]
-        public Dictionary<string, string> CODE1G
-        {
-            get
-            {
-                if (CauseOfDeathConditionG != null && CauseOfDeathConditionG.Code != null)
+                if (ConditionContributingToDeath != null && ConditionContributingToDeath.Value != null)
                 {
-                    return CodeableConceptToDict(CauseOfDeathConditionG.Code);
+                    return (CodeableConceptToDict((CodeableConcept)ConditionContributingToDeath.Value))["text"];
                 }
-                return EmptyCodeableDict();
+                return null;
             }
             set
             {
-               if (CauseOfDeathConditionG == null)
-                {
-                    CauseOfDeathConditionG = CauseOfDeathCondition(6);
-                }
-                if (CauseOfDeathConditionG.Code != null)
+                if (ConditionContributingToDeath != null)
                 {
-                    CodeableConcept code = DictToCodeableConcept(value);
-                    code.Text = CauseOfDeathConditionG.Code.Text;
-                    CauseOfDeathConditionG.Code = code;
+                    ConditionContributingToDeath.Value = new CodeableConcept(null, null, null, value);
                 }
```

Comment:
are these ever set to the code, system, and display? I thought there were some in the examples that had those values set but can't track where those are handled in the code

Reply:
The CONTRIBUTING value is mapped to the .text field.
NCHS doesn't want coded input, but it could be provided and ignored.